src/execution/interpreter_loop.rs

```diff
@@ -75,30 +94,36 @@ pub(super) fn run<H: HookSet>(
 
                 trace!("end of function reached, returning to previous stack frame");
                 wasm.pc = maybe_return_address;
-            }
-            RETURN => {
-                trace!("returning from function");
-
-                let func_to_call_idx = stack.current_stackframe().func_idx;
+                stp = maybe_return_stp;
 
-                let func_to_call_inst = store.funcs.get(func_to_call_idx).unwrap_validated();
-                let func_to_call_ty = types.get(func_to_call_inst.ty).unwrap_validated();
-
-                let ret_vals = stack
-                    .pop_tail_iter(func_to_call_ty.returns.valtypes.len())
-                    .collect::<Vec<_>>();
-                stack.clear_callframe_values();
+                current_sidetable = &store
+                    .funcs
+                    .get(stack.current_stackframe().func_idx)
+                    .unwrap_validated()
+                    .sidetable;
+            }
+            BR_IF => {
+                wasm.read_var_u32().unwrap_validated();
 
-                for val in ret_vals {
-                    stack.push_value(val);
-                }
+                let c: i32 = stack.pop_value(ValType::NumType(NumType::I32)).into();
 
-                if stack.callframe_count() == 1 {
-                    break;
+                if c != 0 {
+                    do_sidetable_control_transfer(&mut wasm, stack, &mut stp, current_sidetable);
+                } else {
+                    stp += 1;
                 }
-
-                trace!("end of function reached, returning to previous stack frame");
-                wasm.pc = stack.pop_stackframe();
+            }
+            BR => {
+                //skip n of BR n
+                wasm.read_var_u32().unwrap_validated();
+                do_sidetable_control_transfer(&mut wasm, stack, &mut stp, current_sidetable);
+            }
+            BLOCK | LOOP => {
+                BlockType::read_unvalidated(&mut wasm);
```

Comment:
Arw we doing anything with this return value?

Reply:
No, I simply parsed it to skip it since blocktype parsing is a bit convoluted, but I can write a skip method as well.